tests/kubesonic/test_k8s_cleanup.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import pytest
+
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CLEANUP_CONTAINER_IMAGE = "kubesonic-cleanup"
+CLEANUP_CONTAINER_NAME = "k8s_cleanup"
+
+
+def pytest_generate_tests(metafunc):
+    if "containers" in metafunc.fixturenames:
+        metafunc.parametrize("containers", [metafunc.config.getoption("containers")], scope="module")
+
+
+def test_k8s_cleanup(duthosts, rand_one_dut_hostname, containers):
+    """
+    Test the cleanup of kubesonic containers
+    """
+    # Check if the test is called from container upgrade test
+    if not containers:
+        pytest.skip("Skipping test as this test is not called from container upgrade test")
+
+    if CLEANUP_CONTAINER_IMAGE not in containers:
```

Comment:
Perhaps you can check docker images or running dockers to see if kubesonic cleanup container was installed directly instead of seeing if its a parameter to a test? Does this satisfy your use case? Container upgrade test should have installed this container before you run this test.